Mathlib/Combinatorics/SimpleGraph/Connectivity.lean

```diff
@@ -2406,6 +2406,31 @@ theorem set_walk_length_succ_eq (u v : V) (n : ℕ) :
 
 variable (G) [DecidableEq V]
 
+theorem set_walk_length_two_bij_commonNeighbors (u v : V) :
```

Comment:
As discussed on Discord, I think it's better to have:
```lean
@[simps]
def subtypeWalkLengthEqTwoEquivcommonNeighbors (u v : V) :
    {p : G.Walk u v // p.length = 2} ≃ G.commonNeighbors u v where
  toFun p := ⟨p.val.getVert 1, match p with | ⟨.cons _ (.cons _ .nil), hp⟩ => ⟨‹G.Adj u _›, ‹G.Adj _ v›.symm⟩⟩
  invFun w := ⟨w.prop.1.toWalk.concat w.prop.2.symm, rfl⟩
  left_inv | ⟨.cons _ (.cons _ .nil), hp⟩ => by rfl
  right_inv | ⟨w, ⟨a, b⟩⟩ => Subtype.ext rfl
```

Reply:
Done.